### ISSUE TITLE
add react to peerDependencies

### ISSUE THREAD

**Issue by pocesar** [rt2zz/redux-persist#646]
Right now, using JSPM, a forced map is needed to resolve `import React, { PureComponent } 'react'` because it doesn't list react as a dependency (or peer).

**Comment by rt2zz** [comment]
hm, I am not sure how best to deal with this. React is not really a dep *nor* peer dep of redux-persist. 

We simply ship react integration as a convenience. Perhaps we should also publish a `redux-persist-react` module?

**Comment by pocesar** [comment]
yeah, I guess. since the primary usage isn't for React, would make sense to be a separate module

**Comment by yordis** [comment]
```
node_modules/redux-persist/src/index.d.ts(69,46): error TS2307: Cannot find module 'react'.
node_modules/redux-persist/src/index.d.ts(91,38): error TS2304: Cannot find name 'React'.
```

Related to Typescript ....

**Comment by rt2zz** [comment]
I see so I suspect we need to move the react type defs out of src/index.d.ts and into `integration/react`. We already have a proxy module there so I think we can add a react specific type definition in that folder w a reference in `integration/react/package.json`

**Comment by lestard** [comment]
We are using redux with angular and this prevents us from using redux-persist.
It would be great to have a UI-agnostic version of redux-persist.
